src/sysext/sysext.c

```diff
@@ -2268,6 +2269,49 @@ static int vl_method_refresh(sd_varlink *link, sd_json_variant *parameters, sd_v
         return sd_varlink_reply(link, NULL);
 }
 
+static int parse_version(const char *filename, char **ret) {
+        const char *start, *ext_suffix;
+        char *version = NULL;
+
+        assert(filename);
+        assert(ret);
+
+        /* Find the last occurrences of '_'. This will mark the start of the extension version. */
+        start = strrchr(filename, '_');
+
+        /* If no underscore found, assign "none" and return success.
+        * This is for cases where there are no versions in the filename like 'image.raw'. */
+        if (!start)
+                goto finish;
+        start ++;
+
+        /* We are only looking for extension images here, because for directories, we want a no-op*/
+        ext_suffix = ENDSWITH_SET(start, ".sysext.raw", ".confext.raw", ".raw");
+
+        /* If no valid extension filename suffix is found, assign "none" and return success. */
+        if (!ext_suffix)
+                goto finish;
+
+        size_t version_length = ext_suffix - start;
+        if (version_length == 0)
+                goto finish;
+
+        /* Extract the version substring */
+        version = strndup(start, version_length);
+        if (!version)
+                return -ENOMEM;
+        *ret = version;
+
+        return 0;
+
+finish:
+        version = strdup("none");
```

Comment:
if an image doesn't carry version info we should insert this as NULL into the table, which will then use the "ersatz" character for the field.